Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/DefaultMousePointer.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+using UInput = UnityEngine.Input;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Internal Touch Pointer Implementation.
+    /// </summary>
+    public class DefaultMousePointer : BaseControllerPointer, IMixedRealityMousePointer
+    {
+        protected float timeoutTimer = 0.0f;
+
+        private bool isInteractionEnabled = false;
+
+        private bool cursorWasDisabledOnDown = false;
+
+        protected bool isDisabled = true;
+
+        #region IMixedRealityMousePointer Implementation
+
+        [SerializeField]
+        [Tooltip("Should the mouse cursor be hidden when no active input is received?")]
+        private bool hideCursorWhenInactive = true;
+
+        /// <inheritdoc />
+        public bool HideCursorWhenInactive => hideCursorWhenInactive;
+
+        [SerializeField]
+        [Range(0.01f, 1f)]
+        [Tooltip("What is the movement threshold to reach before un-hiding mouse cursor?")]
+        private float movementThresholdToUnHide = 0.1f;
+
+        /// <inheritdoc />
+        public float MovementThresholdToUnHide => movementThresholdToUnHide;
+
+        [SerializeField]
+        [Range(0f, 10f)]
+        [Tooltip("How long should it take before the mouse cursor is hidden?")]
+        private float hideTimeout = 3.0f;
+
+        /// <inheritdoc />
+        public float HideTimeout => hideTimeout;
+
+        #endregion IMixedRealityMousePointer Implementation
+
+        #region IMixedRealityPointer Implementation
+
+        /// <inheritdoc />
+        public override bool IsInteractionEnabled => isInteractionEnabled;
+
+        private IMixedRealityController controller;
+        private Vector2 mousePosition;
+
+        /// <inheritdoc />
+        public override IMixedRealityController Controller
+        {
+            get { return controller; }
+            set
+            {
+                controller = value;
+                TrackingState = TrackingState.NotApplicable;
+
+                if (controller != null && gameObject != null)
+                {
+                    InputSourceParent = controller.InputSource;
+                    Handedness = controller.ControllerHandedness;
+                    gameObject.name = "Spatial Mouse Pointer";
```

Comment:
this should probably be Screen Space Mouse Pointer in case of the 2d one?

Reply:
each subclass now defines the name via overriden property